sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -459,6 +460,29 @@ object SQLConf {
     .intConf
     .createWithDefault(4096)
 
+  val IS_PARQUET_PARTITION_ADAPTIVE_ENABLED = buildConf("spark.sql.parquet.adaptiveFileSplit")
```

Comment:
This configuration doesn't look specific to parquet anymore.

Reply:
@HyukjinKwon  Yes, setting spark.sql.files.maxPartitionBytes explicitly do works. But for ad-hoc query, the selected columns are different for different queries, and it's not convenient or event impossible for users to set different maxPartitionBytes for different queries. And for general user (non advanced user), it's not easy for them to calculate a proper value of maxPartitionBytes.